### ISSUE TITLE
训练任务莫名其妙被杀死，没有任何错误提示

### ISSUE THREAD

**Issue by qiyueliuhuo** [PaddlePaddle/PaddleOCR#4540]
python3 tools/train.py -c configs/e2e/e2e_r50_vd_pg.yml -o Global.load_static_weights=False 
/home/andy/workplace/softwares/anaconda3/envs/PaddleOCR/lib/python3.7/site-packages/skimage/morphology/_skeletonize.py:241: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  0, 1, 1, 0, 0, 1, 0, 0, 0], dtype=np.bool)
/home/andy/workplace/softwares/anaconda3/envs/PaddleOCR/lib/python3.7/site-packages/skimage/morphology/_skeletonize.py:256: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=np.bool)
[2021/11/03 20:51:24] root INFO: Architecture : 
[2021/11/03 20:51:24] root INFO:     Backbone : 
[2021/11/03 20:51:24] root INFO:         layers : 50
[2021/11/03 20:51:24] root INFO:         name : ResNet
[2021/11/03 20:51:24] root INFO:     Head : 
[2021/11/03 20:51:24] root INFO:         name : PGHead
[2021/11/03 20:51:24] root INFO:     Neck : 
[2021/11/03 20:51:24] root INFO:         name : PGFPN
[2021/11/03 20:51:24] root INFO:     Transform : None
[2021/11/03 20:51:24] root INFO:     algorithm : PGNet
[2021/11/03 20:51:24] root INFO:     model_type : e2e
[2021/11/03 20:51:24] root INFO: Eval : 
[2021/11/03 20:51:24] root INFO:     dataset : 
[2021/11/03 20:51:24] root INFO:         data_dir : ./train_data/
[2021/11/03 20:51:24] root INFO:         label_file_list : ['./train_data/ocrsetv3/Label_test.txt']
[2021/11/03 20:51:24] root INFO:         name : PGDataSet
[2021/11/03 20:51:24] root INFO:         transforms : 
[2021/11/03 20:51:24] root INFO:             DecodeImage : 
[2021/11/03 20:51:24] root INFO:                 channel_first : False
[2021/11/03 20:51:24] root INFO:                 img_mode : RGB
[2021/11/03 20:51:24] root INFO:             E2ELabelEncodeTest : None
[2021/11/03 20:51:24] root INFO:             E2EResizeForTest : 
[2021/11/03 20:51:24] root INFO:                 max_side_len : 768
[2021/11/03 20:51:24] root INFO:             NormalizeImage : 
[2021/11/03 20:51:24] root INFO:                 mean : [0.485, 0.456, 0.406]
[2021/11/03 20:51:24] root INFO:                 order : hwc
[2021/11/03 20:51:24] root INFO:                 scale : 1./255.
[2021/11/03 20:51:24] root INFO:                 std : [0.229, 0.224, 0.225]
[2021/11/03 20:51:24] root INFO:             ToCHWImage : None
[2021/11/03 20:51:24] root INFO:             KeepKeys : 
[2021/11/03 20:51:24] root INFO:                 keep_keys : ['image', 'shape', 'polys', 'texts', 'ignore_tags', 'img_id']
[2021/11/03 20:51:24] root INFO:     loader : 
[2021/11/03 20:51:24] root INFO:         batch_size_per_card : 1
[2021/11/03 20:51:24] root INFO:         drop_last : False
[2021/11/03 20:51:24] root INFO:         num_workers : 2
[2021/11/03 20:51:24] root INFO:         shuffle : False
[2021/11/03 20:51:24] root INFO: Global : 
[2021/11/03 20:51:24] root INFO:     cal_metric_during_train : False
[2021/11/03 20:51:24] root INFO:     character_dict_path : ppocr/utils/ic15_dict.txt
[2021/11/03 20:51:24] root INFO:     character_type : EN
[2021/11/03 20:51:24] root INFO:     checkpoints : ./pretrain_models/train_step1/best_accuracy
[2021/11/03 20:51:24] root INFO:     debug : False
[2021/11/03 20:51:24] root INFO:     distributed : False
[2021/11/03 20:51:24] root INFO:     epoch_num : 600
[2021/11/03 20:51:24] root INFO:     eval_batch_step : [0, 1000]
[2021/11/03 20:51:24] root INFO:     infer_img : None
[2021/11/03 20:51:24] root INFO:     load_static_weights : False
[2021/11/03 20:51:24] root INFO:     log_smooth_window : 20
[2021/11/03 20:51:24] root INFO:     max_text_length : 50
[2021/11/03 20:51:24] root INFO:     max_text_nums : 30
[2021/11/03 20:51:24] root INFO:     pretrained_model : None
[2021/11/03 20:51:24] root INFO:     print_batch_step : 10
[2021/11/03 20:51:24] root INFO:     save_epoch_step : 10
[2021/11/03 20:51:24] root INFO:     save_inference_dir : None
[2021/11/03 20:51:24] root INFO:     save_model_dir : ./output/pgnet_r50_vd_ocrsetv4/
[2021/11/03 20:51:24] root INFO:     save_res_path : ./output/pgnet_r50_vd_ocrsetv4/predicts_pgnet.txt
[2021/11/03 20:51:24] root INFO:     tcl_len : 64
[2021/11/03 20:51:24] root INFO:     use_gpu : True
[2021/11/03 20:51:24] root INFO:     use_visualdl : False
[2021/11/03 20:51:24] root INFO:     valid_set : totaltext
[2021/11/03 20:51:24] root INFO: Loss : 
[2021/11/03 20:51:24] root INFO:     max_text_length : 50
[2021/11/03 20:51:24] root INFO:     max_text_nums : 30
[2021/11/03 20:51:24] root INFO:     name : PGLoss
[2021/11/03 20:51:24] root INFO:     pad_num : 36
[2021/11/03 20:51:24] root INFO:     tcl_bs : 64
[2021/11/03 20:51:24] root INFO: Metric : 
[2021/11/03 20:51:24] root INFO:     character_dict_path : ppocr/utils/ic15_dict.txt
[2021/11/03 20:51:24] root INFO:     gt_mat_dir : ./train_data/total_text/gt
[2021/11/03 20:51:24] root INFO:     main_indicator : f_score_e2e
[2021/11/03 20:51:24] root INFO:     mode : A
[2021/11/03 20:51:24] root INFO:     name : E2EMetric
[2021/11/03 20:51:24] root INFO: Optimizer : 
[2021/11/03 20:51:24] root INFO:     beta1 : 0.9
[2021/11/03 20:51:24] root INFO:     beta2 : 0.999
[2021/11/03 20:51:24] root INFO:     lr : 
[2021/11/03 20:51:24] root INFO:         learning_rate : 0.001
[2021/11/03 20:51:24] root INFO:     name : Adam
[2021/11/03 20:51:24] root INFO:     regularizer : 
[2021/11/03 20:51:24] root INFO:         factor : 0
[2021/11/03 20:51:24] root INFO:         name : L2
[2021/11/03 20:51:24] root INFO: PostProcess : 
[2021/11/03 20:51:24] root INFO:     mode : fast
[2021/11/03 20:51:24] root INFO:     name : PGPostProcess
[2021/11/03 20:51:24] root INFO:     score_thresh : 0.5
[2021/11/03 20:51:24] root INFO: Train : 
[2021/11/03 20:51:24] root INFO:     dataset : 
[2021/11/03 20:51:24] root INFO:         data_dir : ./train_data/
[2021/11/03 20:51:24] root INFO:         label_file_list : ['./train_data/ocrsetv3/Label.txt']
[2021/11/03 20:51:24] root INFO:         name : PGDataSet
[2021/11/03 20:51:24] root INFO:         ratio_list : [1.0]
[2021/11/03 20:51:24] root INFO:         transforms : 
[2021/11/03 20:51:24] root INFO:             DecodeImage : 
[2021/11/03 20:51:24] root INFO:                 channel_first : False
[2021/11/03 20:51:24] root INFO:                 img_mode : BGR
[2021/11/03 20:51:24] root INFO:             E2ELabelEncodeTrain : None
[2021/11/03 20:51:24] root INFO:             PGProcessTrain : 
[2021/11/03 20:51:24] root INFO:                 batch_size : 8
[2021/11/03 20:51:24] root INFO:                 max_text_size : 512
[2021/11/03 20:51:24] root INFO:                 min_crop_size : 24
[2021/11/03 20:51:24] root INFO:                 min_text_size : 4
[2021/11/03 20:51:24] root INFO:             KeepKeys : 
[2021/11/03 20:51:24] root INFO:                 keep_keys : ['images', 'tcl_maps', 'tcl_label_maps', 'border_maps', 'direction_maps', 'training_masks', 'label_list', 'pos_list', 'pos_mask']
[2021/11/03 20:51:24] root INFO:     loader : 
[2021/11/03 20:51:24] root INFO:         batch_size_per_card : 8
[2021/11/03 20:51:24] root INFO:         drop_last : True
[2021/11/03 20:51:24] root INFO:         num_workers : 2
[2021/11/03 20:51:24] root INFO:         shuffle : True
[2021/11/03 20:51:24] root INFO: train with paddle 2.1.3 and device CUDAPlace(0)
[2021/11/03 20:51:24] root INFO: Initialize indexs of datasets:['./train_data/ocrsetv3/Label.txt']
[2021/11/03 20:51:24] root INFO: Initialize indexs of datasets:['./train_data/ocrsetv3/Label_test.txt']
W1103 20:51:24.253569 602766 device_context.cc:404] Please NOTE: device: 0, GPU Compute Capability: 8.6, Driver API Version: 11.4, Runtime API Version: 11.2
W1103 20:51:24.261466 602766 device_context.cc:422] device: 0, cuDNN Version: 8.1.
[2021/11/03 20:51:28] root INFO: resume from ./pretrain_models/train_step1/best_accuracy
[2021/11/03 20:51:28] root INFO: train dataloader has 848 iters
[2021/11/03 20:51:28] root INFO: valid dataloader has 1500 iters
[2021/11/03 20:51:28] root INFO: During the training process, after the 0th iteration, an evaluation is run every 1000 iterations
[2021/11/03 20:51:28] root INFO: Initialize indexs of datasets:['./train_data/ocrsetv3/Label.txt']
已杀死
没有任何错误提示，换了好几个服务器环境
有人遇到吗

**Comment by qiyueliuhuo** [comment]
换了两三个都会出现这种情况

上面已杀死是程序提示出来的，但没说具体信息

**Comment by qiyueliuhuo** [comment]
可能找到原因了
可能是内存不够吧

重启电脑就好了

这个问题的原因是内存泄露么？

**Comment by tink2123** [comment]
可能是内存不足，一次性读入一个batch的数据导致直接kill，清理内存或者减小batch_size应该都可以解决这个问题
